### DISCUSSION TITLE
Handle errors (Violation errors .. etc)

### DISCUSSION THREAD

**Discussion by asaadawey** [prisma/prisma#3264] [Q&A] [answered]
What is the proper way to handle errors ?
ex constraint violation errors 

here is a code : 
```
         const user = await ctx.prisma.users.create({
          data: {
            FIRST_NAME,
            EMAIL,
            PASSWORD: hashedPassword,
            IS_ACTIVE,
            USERNAME,
            CREATED_BY
          },
        })
```

How to pass custom error if there is constraint violation ?

**Comment by ryands17** [comment, marked as answer]
Hey @asaadawey 👋 
This question is being asked a lot, and currently there is no great way to handle this, for now you could check for the *Unique constraint* violation part in the error message and beased on that perform any operation that you want to

For e.g. I am inserting two rows with the same username, and in turn violating the unique constraint.

```ts
async function main() {
  await prisma.user.create({
    data: {
      username: 'abc',
    },
  })

  await prisma.user.create({
    data: {
      username: 'abc',
    },
  })
}

ain()
  .catch((err) => {
    console.log(err.message.slice(err.message.indexOf('Unique constraint')))
  })
  .finally(async () => {
    await prisma.$disconnect()
  })
```

This prints out the following message in the console:
![image](https://user-images.githubusercontent.com/19697099/89872965-58a6cf00-dbd7-11ea-93fc-4155b623a841.png)

Let me know if this helps :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
